CSharp/MineStat/MineStat.cs

```diff
@@ -154,43 +165,185 @@ public MineStat(string address, ushort port, int timeout = DefaultTimeout, SlpPr
         case SlpProtocol.Json:
           RequestWrapper(RequestWithJsonProtocol);
           break;
+        case SlpProtocol.Bedrock_Raknet:
+          RequestWithRaknetProtocol();
+          break;
         case SlpProtocol.Automatic:
           break;
         default:
           throw new ArgumentOutOfRangeException(nameof(protocol), "Invalid SLP protocol specified for parameter 'protocol'");
       }
-      
+
       // If a protocol was chosen manually, return
       if (protocol != SlpProtocol.Automatic)
       {
         return;
       }
-      
+
       // The order of protocols here is (sadly) important.
       // Some server versions (1.3, 1.4) seem to have trouble with newer protocols and stop responding for a few seconds.
       // If, for example, the ext.-legacy protocol triggers this problem, the following connections are dropped/reset
       // even if they would have worked individually/normally.
       // For more information, see https://github.com/FragLand/minestat/issues/70
       //
-      // 1.: Legacy (1.4, 1.5)
-      // 2.: Beta (b1.8-rel1.3)
-      // 3.: Extended Legacy (1.6)
-      // 4.: JSON (1.7+)
-      
-      var result = RequestWrapper(RequestWithLegacyProtocol);
-      
+      // 1.: Raknet (Bedrock)
+      // 2.: Legacy (1.4, 1.5)
+      // 3.: Beta (b1.8-rel1.3)
+      // 4.: Extended Legacy (1.6)
+      // 5.: JSON (1.7+)
+
+      var result = RequestWithRaknetProtocol();
+      if (result == ConnStatus.Connfail || result == ConnStatus.Success)
+        return;
+
+      result = RequestWrapper(RequestWithLegacyProtocol);
+
       if (result != ConnStatus.Connfail && result != ConnStatus.Success)
       {
         result = RequestWrapper(RequestWithBetaProtocol);
-        
-        if (result != ConnStatus.Connfail)
-          result = RequestWrapper(RequestWithExtendedLegacyProtocol);
+      }
+      if (result != ConnStatus.Connfail)
+        result = RequestWrapper(RequestWithExtendedLegacyProtocol);
 
-        if (result != ConnStatus.Connfail && result != ConnStatus.Success)
-          RequestWrapper(RequestWithJsonProtocol);
+      if (result != ConnStatus.Connfail /* && result != ConnStatus.Success */)
+        RequestWrapper(RequestWithJsonProtocol);
+    }
+    
+    /// <summary>
+    /// Function for stripping all formatting codes from a motd.
+    /// </summary>
+    /// <returns>string with the stripped motd</returns>
+    static private string strip_motd_formatting(string? rawmotd)
+    {
+      return Regex.Replace(rawmotd, @"[^\u0000-\u007F]+[a-zA-Z0-9]", string.Empty);
```

Comment:
Why are you searching for all non-ASCII-characters? The string returned by the server is UTF8 (or UTF16) encoded and can contain any of those characters, e.g. Chinese or language-specifics like the German umlauts (äöü), which will trigger this regex. Did you find special cases where other dividiers then `§` were used?
![grafik](https://user-images.githubusercontent.com/24863329/178063455-e2194233-ad4b-4832-b610-d346ddb56b0c.png)

Reply:
No, it's just an error on my part. Just matching § with `@"\u00A7+[a-zA-Z0-9]"` works!